cpp/oneapi/dal/algo/kmeans_init/backend/cpu/compute_kernel_dense.cpp

```diff
@@ -43,9 +45,17 @@ static compute_result<Task> call_daal_kernel(const context_cpu& ctx,
     const std::int64_t column_count = data.get_column_count();
     const std::int64_t cluster_count = desc.get_cluster_count();
 
+    std::int64_t trial_count = desc.get_local_trials_count();
+    if (trial_count == -1)
+    {
+        const auto additional = std::log(cluster_count);
+        trial_count = 2 + std::int64_t(additional);
+    }
```

Comment:
Please add the comment describing the logic behind this. Maybe the link to Sklearn docs.